packages/front-end/components/Settings/SnowflakeForm.tsx

```diff
@@ -35,19 +40,93 @@ const SnowflakeForm: FC<{
           onChange={onParamChange}
         />
       </div>
+
       <div className="form-group col-md-12">
-        <label>Password</label>
-        <input
-          type="text"
-          className="form-control password-presentation"
+        <label>Authentication Method</label>
+        <select
+          className="form-control"
           autoComplete="off"
-          name="password"
-          required={!existing}
-          value={params.password || ""}
-          onChange={onParamChange}
-          placeholder={existing ? "(Keep existing)" : ""}
-        />
+          name="authMethod"
+          value={params.authMethod ?? "password"}
+          onChange={(e) => onManualParamChange("authMethod", e.target.value)}
+        >
+          <option value="password">Password</option>
+          <option value="key-pair">Key Pair</option>
+        </select>
       </div>
+
+      {authMethod === "password" ? (
+        <div className="form-group col-md-12">
+          <label>Password</label>
+          <input
+            type="text"
+            className="form-control password-presentation"
+            autoComplete="off"
+            name="password"
+            required={!existing || authMethod !== originalAuthMethod}
+            value={params.password || ""}
+            onChange={onParamChange}
+            placeholder={
+              existing && authMethod === originalAuthMethod
+                ? "(Keep existing)"
+                : ""
+            }
+          />
+        </div>
+      ) : null}
+
+      {authMethod === "key-pair" ? (
+        <>
+          <div className="form-group col-md-12">
+            <label>Private Key File</label>
+            <input
```

Comment:
TODO: Create a custom component so we can show `(Existing)` as well for users who already have a private key uploaded.